tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -484,12 +485,23 @@ public void processTonyConfResources(Configuration tonyConf, FileSystem fs) thro
             }
           } else {
             File tmpDir = Files.createTempDir();
-            Utils.zipFolder(Paths.get(resource), Paths.get(tmpDir.getAbsolutePath(), file.getName()));
-            Utils.uploadFileAndSetConfResources(appResourcesPath,
-                    new Path(trimmedResource),
-                    new Path(trimmedResource).getName(),
-                    tonyConf,
-                    fs, LocalResourceType.ARCHIVE, jobName);
+            tmpDir.deleteOnExit();
+            try {
+              java.nio.file.Path dest = Paths.get(tmpDir.getAbsolutePath(), file.getName());
+              Utils.zipFolder(Paths.get(resource), dest);
+              Utils.uploadFileAndSetConfResources(appResourcesPath,
+                  new Path(dest.toString()),
+                  new Path(dest.toString()).getName(),
+                  tonyConf,
+                  fs, LocalResourceType.ARCHIVE, jobName);
```

Comment:
I know you didn't name the variable originally, but could you rename `jobName` and `jobNames` to `resourceKey` and `resourceKeys`?

Reply:
updated, thanks